crates/emitter/src/ast_emitter.rs

```diff
@@ -430,23 +432,18 @@ impl<'alloc> AstEmitter<'alloc> {
         right: &Expression,
     ) -> Result<(), EmitError> {
         self.emit_expression(left)?;
-        let mut jumplist: Vec<BytecodeOffset> = Vec::with_capacity(1);
-        self.emit_jump(operator, &mut jumplist)?;
-        self.emit.jump_target();
+        let forward_jump = ForwardJumpEmitter {
+            offset: self.emit.bytecode_offset(),
```

Comment:
I think, `offset` should be retrieved automatically by helper classes, when emitting the forward jump.
(given that, passing another value will result in a bug)

then, with this pattern, you cannot omit fields in the caller, so we need some workaround.

How about creating 2 structs for:
  * A) emits forward jump
  * B) patches the jump

and use like the following?

```
let b: B = A { jump: ... }.emit(self)?;
...
b.patch(self);
```

given that `offset` is used only by B, and `jump` is used only by A, separating them should work.
also, by making B's field and constructor private, it's hard to mis-use.

Then, thinking about `jump` part,
given that it's supposed to emit single jump opcode, I think the helper can receive single `enum`,
instead of closure, and emit jump op depending on the `enum` (so, copy `emit_jump` there, with a new `enum` for all jump kinds).
so that the caller doesn't have to touch `InstructionWriter` at all, while emitting jump related code.